src/dotnet/commands/dotnet-add/dotnet-add-proj/Program.cs

```diff
@@ -104,9 +105,93 @@ private void AddProject(SlnFile slnFile, string projectPath)
                 };
 
                 slnFile.Projects.Add(slnProject);
+
+                EnsureBuildConfigurationsAreValidAndComplete(slnFile);
+
                 Reporter.Output.WriteLine(
                     string.Format(CommonLocalizableStrings.ProjectAddedToTheSolution, projectPath));
             }
         }
+
+        private void EnsureBuildConfigurationsAreValidAndComplete(SlnFile slnFile)
+        {
+            HashSet<string> _buildConfigurations = new HashSet<string>()
+            {
+                "Debug",
```

Comment:
They can, I'm changing the algorithm again and will create a new PR. Basically I won't be trying to fix up a broken sln anymore. The goal is just to not mess up a valid one.